app/memorize.py

```diff
@@ -0,0 +1,137 @@
+from functools import wraps
+import hashlib
+import pickle
+
+from wrapt import ObjectProxy
+
+from memcached import memcached
+
+
+class CacheMissError(RuntimeError):
+    """
+    Exception thrown when memorized object is not found in cache and
+    error_on_miss is set to True.
+    """
+    pass
+
+
+class MemorizedObject(ObjectProxy):
```

Comment:
I see what you did there (memoize, memorize). /me _golf claps_

![](https://media.giphy.com/media/4la2AJWCtl8Pu/giphy.gif)
![](https://media.giphy.com/media/IUZtGhVO8hZ6w/giphy.gif)
![](https://media.giphy.com/media/wijMRo7UZXSqA/giphy.gif)